### PR TITLE
Cleans up command options package

### PR DESCRIPTION
**Pull request by JoshVanL**
Managing options for the kube-oidc-proxy has become quite unwieldy which makes it difficult to work on and add new flags. This PR updates the options package to return a single options struct which provides a much better interface for the consuming package. This PR also moves more into the options package where it makes sense which cleans things up a bit more.

/assign @simonswine

### CODE REVIEW

**Comment by simonswine** [comment]
Needs rebase

/assign @JoshVanL 
/unassign

File: cmd/app/options/options.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+func New() *Options {
+	nfs := new(cliflag.NamedFlagSets)
+
+	// Add flags to command sets
+	return &Options{
+		OIDCAuthentication: NewOIDCAuthenticationOptions(nfs),
+		SecureServing:      NewSecureServingOptions(nfs),
+		Client:             NewClientOptions(nfs),
+		App:                NewKubeOIDCProxyOptions(nfs),
```
Comment (simonswine) [inline cmd/app/options/options.go]: I would prefer if we could have the KubeOIDCProxy flags first in the help output (as we had it before). I think if we move that line to the top of the struct we will see it first.

File: pkg/util/errors.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
+// Copyright Jetstack Ltd. See LICENSE for details.
+package util
+
+import (
+	"errors"
+)
+
+// JoinErrors will join a slice of errors into a single error, pretty
+// printed.
+func JoinErrors(errs []error) error {
```
Comment (simonswine) [inline pkg/util/errors.go]: Consider using https://github.com/kubernetes/apimachinery/blob/master/pkg/util/errors/errors.go#L41

**Comment by JoshVanL** [comment]
Think I addressed your comments
/unassign
/assign @simonswine

**Comment by simonswine** [comment]
Nice on, thanks 

/lgtm

**Pull request merged**